src/coreclr/jit/llvm.cpp

```diff
@@ -1896,7 +1857,277 @@ void Llvm::ConvertShadowStackLocalNode(GenTreeLclVarCommon* node)
+    std::vector<OperandArgNum> sortedArgs = std::vector<OperandArgNum>(argCount);
+    OperandArgNum*             sortedData = sortedArgs.data();
+
+    GenTreeCall::Use* oldArgs = callNode->gtCallLateArgs;
+    GenTreeCall::Use* lastArg;
+    GenTreeCall::Use* insertReturnAfter;
+    GenTreeCall::Use* callThisArg = callNode->gtCallThisArg;
+
+    callNode->ResetArgInfo();
+    callNode->gtCallThisArg = nullptr;
+    argCount                = argInfo->ArgCount();
+    argTable                = argInfo->ArgTable();
+
+    // set up the callee shadowstack, creating a temp and the PUTARG
+    GenTreeLclVar* shadowStackVar = _compiler->gtNewLclvNode(_shadowStackLclNum, TYP_I_IMPL);
+    GenTreeIntCon* offset         = _compiler->gtNewIconNode(_shadowStackLocalsSize,
+                                                     TYP_I_IMPL); // TODO-LLVM: possible performance benefit: when
+                                                                  // _shadowStackLocalsSize == 0, then omit the GT_ADD
+    GenTree* calleeShadowStack = _compiler->gtNewOperNode(GT_ADD, TYP_I_IMPL, shadowStackVar, offset);
+
+    // store in the temp
+    unsigned calleeShadowStackTemp = _compiler->lvaGrabTemp(
+        false DEBUGARG("callee shadowstack")); // used to create the store locations for args on shadow stack
+    LclVarDsc* calleeShadowStackVarDsc = _compiler->lvaGetDesc(calleeShadowStackTemp);
+    calleeShadowStackVarDsc->lvType    = TYP_I_IMPL;
+    GenTree* storeShadowStack          = _compiler->gtNewStoreLclVar(calleeShadowStackTemp, calleeShadowStack);
+
+    // create the PUTARG
+    GenTree*           lclCalleeShadowStack = _compiler->gtNewLclvNode(calleeShadowStackTemp, TYP_I_IMPL);
+    GenTreePutArgType* putArg =
+        _compiler->gtNewOperNode(GT_PUTARG_TYPE, TYP_I_IMPL, lclCalleeShadowStack)->AsPutArgType();
+    putArg->gtCorInfoType = CORINFO_TYPE_PTR;
+    putArg->gtClsHnd      = nullptr;
+
+    callNode->gtCallArgs     = _compiler->gtNewCallArgs(putArg);
+    lastArg                  = callNode->gtCallArgs;
+    insertReturnAfter        = lastArg; // add the return slot after the shadow stack arg
+    callNode->gtCallLateArgs = nullptr;
+
+    CurrentRange().InsertBefore(callNode, shadowStackVar, offset, calleeShadowStack, storeShadowStack);
+    CurrentRange().InsertAfter(storeShadowStack, lclCalleeShadowStack, putArg);
+
+    lastArg = lowerCallReturn(callNode, calleeSigInfo, insertReturnAfter);
+
+    for (unsigned i = 0; i < argCount; i++)
+    {
+        fgArgTabEntry* curArgTabEntry = argTable[i];
+        unsigned int   argNum         = curArgTabEntry->argNum;
+        OperandArgNum  opAndArg       = {argNum, curArgTabEntry->GetNode()};
+        sortedData[argNum]            = opAndArg;
+    }
+
+    CORINFO_ARG_LIST_HANDLE sigArgs = calleeSigInfo.args;
+
+    for (OperandArgNum opAndArg : sortedArgs)
+    {
+        if (opAndArg.operand->IsArgPlaceHolderNode())
+        {
+            thisArgNum++;
+            continue;
+        }
```

Comment:
Hmm, not sure I understand this logic: `opAndArg.operand` should never be a placeholder (since it is obtained from `curArgTabEntry->GetNode()`).

Reply:
Its not needed, I'll delete it.